### ISSUE TITLE
[POP-404] Deprecation check failed

### ISSUE THREAD

**Issue by realsarm** [derailed/popeye#181]
Hi
I'm getting `[POP-404] Deprecation check failed. Unable to assert resource version` on ingresses on GKE.
What is the problem?

**Comment by survivant** [comment]
I have the same question.  Here a Ingress that I have that return a POP-404 message.  
````
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    meta.helm.sh/release-name: acquisition-service-singleton
    meta.helm.sh/release-namespace: dev
    nginx.ingress.kubernetes.io/rewrite-target: /$2
    nginx.ingress.kubernetes.io/ssl-redirect: "false"
  creationTimestamp: "2021-07-07T14:33:12Z"
  generation: 1
  labels:
    app.kubernetes.io/managed-by: Helm
  managedFields:
  - apiVersion: networking.k8s.io/v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:meta.helm.sh/release-name: {}
          f:meta.helm.sh/release-namespace: {}
          f:nginx.ingress.kubernetes.io/rewrite-target: {}
          f:nginx.ingress.kubernetes.io/ssl-redirect: {}
        f:labels:
          .: {}
          f:app.kubernetes.io/managed-by: {}
      f:spec:
        f:rules: {}
    manager: Go-http-client
    operation: Update
    time: "2021-07-07T14:33:12Z"
  - apiVersion: networking.k8s.io/v1beta1
    fieldsType: FieldsV1
    fieldsV1:
      f:status:
        f:loadBalancer:
          f:ingress: {}
    manager: nginx-ingress-controller
    operation: Update
    time: "2021-07-07T14:33:46Z"
  name: acquisition-service
  namespace: dev
  resourceVersion: "55331969"
  uid: fed728c9-ed64-42be-8345-dd4cf3b218f1
spec:
  rules:
  - host: dev.dynamic.cluster109.local
    http:
      paths:
      - backend:
          service:
            name: acquisition-service
            port:
              number: 8080
        path: /acquisition(/|$)(.*)
        pathType: Prefix
status:
  loadBalancer:
    ingress:
    - ip: 10.1.34.54
````

**Comment by AljoschaP** [comment]
I get the same error on almost all resources with K8s version v1.21.2. Its related to the K8s version in combination with the following method:

https://github.com/derailed/popeye/blob/a80f6aeb06be01e0d5384b836239e09c8eadd4aa/internal/sanitize/dp.go#L224

The SelfLink field is deprecated with v1.21.0:

```
	// SelfLink is a URL representing this object.
	// Populated by the system.
	// Read-only.
	//
	// DEPRECATED
	// Kubernetes will stop propagating this field in 1.20 release and the field is planned
	// to be removed in 1.21 release.
	// +optional
	SelfLink string `json:"selfLink,omitempty" protobuf:"bytes,4,opt,name=selfLink"`
```

https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta

**Comment by derailed** [comment]
Fixed v0.9.4!

**Comment by AljoschaP** [comment]
Looks good but container push for v0.9.5 failed I think :see_no_evil:
